src/repo-url.js

```diff
@@ -0,0 +1,57 @@
+const fs = require('fs-extra');
+const path = require('path');
+const shared = require('./utils/shared');
+
+const preload = (content, resourcePath) => {
+  if (!resourcePath.match(/app-extras\.module\.ts$/)) {
+    return content;
+  }
+
+  function generateRepoUrlObject(url) {
+    const githubFilePathRoot = '/tree/master/src/app';
+    const vstsFilePathRoot = '?path=%2Fsrc%2Fapp';
+    const vstsRepoBranchSelector = '&version=GBmaster';
+
+    let result = {};
+
+    if (url.includes('visualstudio')) {
+      result.url = url + vstsFilePathRoot;
+      result.vstsBranchSelector = vstsRepoBranchSelector;
+    } else {
+      result.url = url + githubFilePathRoot;
+    }
+
+    return result;
+  }
+
+  let packageJson;
+
+  try {
+    packageJson = fs.readJsonSync(path.join(process.cwd(), 'package.json'));
+    if (packageJson.repository === undefined) {
+      throw new Error('Package.json must have a repository object and url key to use edit button feature.');
+    }
+  } catch (error) {
+    throw new shared.StachePluginError(error.message);
+  }
+
+  let url = packageJson.repository.url;
+
+  if (url.endsWith('.git')) {
```

Comment:
This might cause problems with VSTS, since their repo URLs do not end with `.git`. What do you think about removing this check entirely, since we can't know the formats for all git services?

For example, the `stache-search-svc` repo URL looks like: https://blackbaud.visualstudio.com/Products/_git/stache-search-svc